ferrocene/ci/scripts/setup-ubuntu.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+# SPDX-License-Identifier: MIT OR Apache-2.0
+# SPDX-FileCopyrightText: The Ferrocene Developers
+set -xeo pipefail
+
+# Ensure we never get asked/prompted, always take the new config
+echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections
+
+# Sometimes (eg the commit job) we want to skip doing anything time intensive like upgrading.
+# Jobs which actually produce artifacts should *always* upgrade as they may otherwise miss critical
+# security upgrades.
+if [[ -z "${SKIP_UPGRADE}" ]]; then
+    echo 'Skipping upgrade, $SKIP_UPGRADE is set. Use caution if any artifacts are produced.'
+else
+    apt update
+    apt upgrade -y
+fi
+
+apt install -y \
+    git \
+    build-essential \
+    ninja-build \
+    zlib1g-dev
+
+if [[ -z "${INSTALL_LLVM}" ]]; then
```

Comment:
I assume I'm having a moment, but doesn't `-z` test if a variable is empty (or of **z**ero length)? So doesn't this install llvm-17-tools and llvm-17-dev if and only if INSTALL_LLVM is of zero length, and won't install llvm tools is INSTALL_LLVM is not of zero length?

Likewise above, doesn't it only run `apt upgrade` if and only if SKIP_UPGRADE is of non-zero length? That seems backwards.

Reply:
Indeed you're right. I don't think I tested this enough.